### ISSUE TITLE
How to use Amplify Storage with S3 static website endpoint?

### ISSUE THREAD

**Issue by jazblueful** [aws-amplify/amplify-js#3257]
__Which Category is your question related to?__
Storage

__What AWS Services are you utilizing?__
Appsync, Lambda, S3, API Gateway

__Provide additional details e.g. code snippets__
Since Amplify.configure doesn't support setting custom endpoint for storage, I configured s3 service directly through aws-sdk:
```javascript
import AWS from 'aws-sdk'
import Amplify from 'aws-amplify'

AWS.config.s3 = {
  endpoint: 'http://my-bucket.s3-website-us-east-1.amazonaws.com'
}

Amplify.configure({
    Storage: {
      AWSS3: {
        bucket: 'my-bucket',
        region: 'us-east-1'
      }
    }
  })
```

When I do Storage.get('key'), Amplify takes this endpoint and adds a bucket name to the url: `http://my-bucket.s3-website-us-east-1.amazonaws.com/`**my-bucket**`/key`, which is wrong and should instead be `http://my-bucket.s3-website-us-east-1.amazonaws.com/key`

I tried explicitly setting s3ForcePathStyle to false in the configuration, which doesn't work.
I also can't omit bucket name from Storage configuration as Amplify will complain when I try to do Storage.get.

In case you are wondering why I insist on using static website hosting for my s3 bucket, I want to use redirection rules of s3 to redirect requests that return with code 404 to image cropper api gateway that uses lambda, which takes the image from s3, crops it, puts it back and redirects the user back to the url, that formerly resulted with 404 and should now return a newly cropped image instead. https://aws.amazon.com/blogs/compute/resize-images-on-the-fly-with-amazon-s3-aws-lambda-and-amazon-api-gateway/

I really hope you can guide me to the right direction.

**Comment by sammartinez** [comment]
@jazblueful In the article you link, it does seem know related to Amplify. Can you please provide a little more detail on what your use case is with Amplify?

**Comment by jazblueful** [comment]
I use Amplify for authentication flow (unrelated in this case) and Storage.put/Storage.get to upload/download files to/from S3

```javascript
Storage.put('key', file);
```

Now there should be an object with the content of file at S3 http://my-bucket.s3-website-us-east-1.amazonaws.com/my-bucket/private/us-east-1:xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/key

```javascript
Storage.get('100x100/key')
```
This will return 404, however redirection rules of static website hosting settings in S3 allow to take all requests that would otherwise return 404 and redirect them, in my case to a cropping lambda function, which takes the image from my-bucket/private/us-east-1:xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/key, crops it, puts it back to my-bucket/private/us-east-1:xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/100x100/key and redirects the caller to http://my-bucket.s3-website-us-east-1.amazonaws.com/my-bucket/private/us-east-1:xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/100x100/key

This should work, except Amplify doesn't allow using S3 buckets configured to act as static website servers.

**Comment by sammartinez** [comment]
@jazblueful That is correct that we don't offer this in Amplify. I'm marking this issue as a feature request

**Comment by renanbronchart** [comment]
How about this issue? I have the same problem. 
Is someone take this feature?
